x-pack/plugins/ml/public/data_frame/common/index.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { DefaultOperator } from 'elasticsearch';
+
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+import { Dictionary } from '../../../common/types/common';
+
+import { DefinePivotExposedState } from '../components/define_pivot/define_pivot_form';
+
+// The display label used for an aggregation e.g. sum(bytes).
+export type Label = string;
+
+// Label object structured for EUI's ComboBox.
+export interface DropDownLabel {
+  label: Label;
+}
+
+// Label object structure for EUI's ComboBox with support for nesting.
+export interface DropDownOption {
+  label: Label;
+  options: DropDownLabel[];
+}
+
+// The internal representation of an aggregation definition.
+type aggName = string;
+type fieldName = string;
+export interface OptionsDataElement {
+  agg: PivotAggSupportedAggs;
+  field: fieldName;
+  formRowLabel: aggName;
+}
+
+export type OptionsDataElementDict = Dictionary<OptionsDataElement>;
+
+export interface SimpleQuery {
+  query: {
+    query_string: {
+      query: string;
+      default_operator: DefaultOperator;
+    };
+  };
+}
+
+// DataFramePreviewRequest
+type PivotGroupBySupportedAggs = 'terms';
+type PivotGroupBy = {
+  [key in PivotGroupBySupportedAggs]: {
+    field: string;
+  }
+};
+type PivotGroupByDict = Dictionary<PivotGroupBy>;
+
+type PivotAggSupportedAggs = 'avg' | 'cardinality' | 'max' | 'min' | 'sum' | 'value_count';
```

Comment:
Is it worth defining these values as e.g an `AGGREGATION_TYPE` enum? I imagine they will be useful elsewhere, such as in anomaly detectors

Reply:
I refactored to use an enum in 6b4b1f4. For now I'll leave it in that file. The supported aggs are specific to data frames, which aggs we support there or in anomaly detectors might differ.